src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -1055,10 +1057,34 @@ PeerImp::onMessage (std::shared_ptr <protocol::TMTransaction> const& m)
             {
                 // For now, be paranoid and have each validator
                 // check each transaction, regardless of source
-                flags |= SF_SIGGOOD;
+                checkSignature = false;
             }
         }
 
+        if (checkSignature)
+        {
+            // Check the signature before handing off to the job queue.
+            auto valid = checkValidity(app_.getHashRouter(), *stx,
```

Comment:
err... we can't be checking the signature on an `io_service` thread! I believe the signature needs to be checked from the `JobQueue`, so the load can be tracked and managed.